R/PackageInheritanceReporter.R

```diff
@@ -0,0 +1,212 @@
+#'       \code{\link[methods:ReferenceClasses]{setRefClass}} is used as the node
+#'       name by this reporter. This is the class name that is used when
+#'       specifying inheritance. The generator object returned by
+#'       \code{\link[methods:ReferenceClasses]{setRefClass}} does not have to be
+#'       assigned and can have a different name.}
+#'       \item{R6 Classes : The name of the generator object in the package
+#'       namespace is used as the node name by this reporter. The generator
+#'       object returned by \code{\link[R6:R6Class]{R6::R6Class}} is what is
+#'       used when specifying inheritance. The name passed as \code{classname}
+#'       passed to \code{\link[R6:R6Class]{R6::R6Class}} can be a different name
+#'       or even NULL.}
+#'  }
+#'
+#'   For more info about R's built-in object-oriented systems, check out the
+#'   relevant chapter in \href{http://adv-r.had.co.nz/OO-essentials.html}{Hadley
+#'   Wickham's \emph{Advanced R}}. For more info about R6, check out their
+#'   \href{https://r6.r-lib.org/index.html}{docs website} or the chapter in
+#'   \href{https://adv-r.hadley.nz/r6.html}{\emph{Advanced R}'s second edition}.
+#'
+#' @section Public Methods:
+#' \describe{
+#'     \item{\code{set_package(pkg_name, pkg_path)}}{
+#'         \itemize{
+#'             \item{Set properties of this reporter. If pkg_name overrides a
+#'                 previously-set package name, any cached data will be removed.}
+#'             \item{\bold{Args:}}{
+#'                 \itemize{
+#'                 \item{\bold{\code{pkg_name}}: String with the name of the package.}
+#'                 \item{\bold{\code{pkg_path}}: Optional directory path to source
+#'                   code of the package. It is used for calculating test coverage.
+#'                   It can be an absolute or relative path.}
+#'                }
+#'             }
+#'         }
+#'     }
+#' }
+#' @importFrom R6 R6Class is.R6Class
+#' @importFrom DT datatable formatRound
+#' @importFrom data.table data.table rbindlist
+#' @importFrom methods is
+#' @export
+InheritanceReporter <- R6::R6Class(
+    "InheritanceReporter",
+    inherit = AbstractGraphReporter,
+
+    public = list(
+        get_summary_view = function(){
+
+            # Calculate network measures if not already done
+            # since we want the node measures in summary
+            invisible(self$network_measures)
+
+            # Create DT for display
+            tableObj <- DT::datatable(
+                data = self$nodes
+                , rownames = FALSE
+                , options = list(
+                    searching = FALSE
+                    , pageLength = 50
+                    , lengthChange = FALSE
+                )
+            )
+            # Round the double columns to three digits for formatting reasons
+            numCols <- names(which(unlist(lapply(tableObj$x$data, is.double))))
+            tableObj <- DT::formatRound(columns = numCols, table = tableObj
```

Comment:
can you change this to our standard indentation?

```
tableObject <- DT::formatRound(
    columns = numCols
    , table = tableObj
    , digits = 3
)
```

Reply:
I'm surprised that this is an assignment :(

does `DT` not support doing this in-place?

Reply:
I had copied and pasted this from another reporter, but I just tested and apparently `DT::formatRound` is not in-place.